specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json

```diff
@@ -0,0 +1,523 @@
+{
+    "swagger": "2.0",
+    "info": {
+      "title": "Guest Diagnostic Settings API",
+      "description": "API to Add/Remove/List Guest Diagnostics Configuration to Azure Resources",
+      "x-ms-code-generation-settings": {
+        "name": "MonitorManagementClient"
+      },
+      "version": "2018-06-01-preview"
+    },
+    "host": "management.azure.com",
+    "schemes": [
+      "https"
+    ],
+    "consumes": [
+      "application/json"
+    ],
+    "produces": [
+      "application/json"
+    ],
+    "security": [
+      {
+        "azure_auth": [
+          "user_impersonation"
+        ]
+      }
+    ],
+    "securityDefinitions": {
+      "azure_auth": {
+        "type": "oauth2",
+        "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+        "flow": "implicit",
+        "description": "Azure Active Directory OAuth2 Flow",
+        "scopes": {
+          "user_impersonation": "impersonate your user account"
+        }
+      }
+    },
+    "paths": {
```

Comment:
Since guestDiagnosticSettings is a top level tracked resource, supporting PATCH is required. Ideally it should support updating all patchable properties. But at the least, it should support updating tags.

Reply:
Done